app/Http/Controllers/Backend/StatisticController.php

```diff
@@ -26,17 +26,30 @@ public static function getGlobalCheckInStats(Carbon $from, Carbon $until): stdCl
             throw new InvalidArgumentException('since cannot be after until');
         }
 
-        return DB::table('train_checkins')
-                 ->join('statuses', 'train_checkins.status_id', '=', 'statuses.id')
-                 ->where('train_checkins.departure', '>=', $from->toIso8601String())
-                 ->where('train_checkins.departure', '<=', $until->toIso8601String())
-                 ->select([
-                              DB::raw('SUM(train_checkins.distance) AS distance'),
-                              DB::raw('SUM(TIMESTAMPDIFF(SECOND, train_checkins.departure,
-                              train_checkins.arrival)) AS duration'),
-                              DB::raw('COUNT(DISTINCT statuses.user_id) AS user_count')
-                          ])
-                 ->first();
+        return self::globalCheckinQuery($from, $until);
+    }
+
+    public static function getGlobalCheckInStatsAllTime(): stdClass {
+        return self::globalCheckinQuery();
+    }
+
+    private static function globalCheckinQuery(?Carbon $from = null, ?Carbon $until = null): stdClass {
+        $query = DB::table('train_checkins');
+
+        if ($from !== null && $until !== null) {
+            $query->where('train_checkins.departure', '>=', $from->toIso8601String())
+                  ->where('train_checkins.departure', '<=', $until->toIso8601String());
+        }
+        $query->selectRaw('SUM(train_checkins.distance) AS distance');
+        $query->selectRaw('COUNT(DISTINCT train_checkins.user_id) AS user_count');
```

Comment:
Ich würd vielleicht eher nur die aktiven User anzeigen, weil die aussagekräftiger sind, als die registrierten.
Dafür müssten wir irgendwo schon in den Statistiken eine Function haben.

Reply:
Das sind alle "aktiven" user. Also alle, die mindestens einen Checkin erstellt haben. :D

Reply:
Hätte noch an eine zeitliche Einschränkung gedacht, aber okay. :D Mir soll's recht sein.